File: advanced/challenge-1/task1/server/models/driver.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Driver extends Model {
    static associate(models) {
      Driver.hasMany(models.Schedule, {
        foreignKey: "driver_id",
        as: "schedules",
      });
    }
  }
  Driver.init(
    {
      first_name: {
        type: DataTypes.STRING,
        allowNull: false,
      },

      last_name: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      social_security_number: {
        type: DataTypes.STRING,
        allowNull: false,
      },

      email: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true,
      },
    },
    {
      sequelize,
      freezeTableName: true,
      tableName: "drivers",
      modelName: "Driver",
      paranoid: true,
    }
  );

  return Driver;
};

File: src/responses/driver.py
import pydantic


class Driver(pydantic.BaseModel):
    class Config:
        orm_mode = True

    id: int
    first_name: str
    last_name: str
    social_security_number: str
    email: str

class TopDriver(Driver):
    class Config:
        orm_mode = True

    schedule_count: int